bundles/org.openhab.binding.miio/src/test/java/org/openhab/binding/miio/internal/ReadmeHelper.java

```diff
@@ -283,14 +312,57 @@ private MiIoBasicDevice findDatabaseEntry(String deviceName) {
         return arrayList;
     }
 
+    private Map<String, String> createI18nEntries() {
+        Map<String, String> i18nEntries = new HashMap<>();
+        String path = "./src/main/resources/database/";
+        File dir = new File(path);
+        File[] filesList = dir.listFiles();
+        for (File file : filesList) {
+            if (file.isFile() && file.getName().toLowerCase().endsWith(".json")) {
+                try {
+                    String key = file.getName().toLowerCase().split("json")[0];
+                    JsonObject deviceMapping = convertFileToJSON(path + file.getName());
+                    Gson gson = new GsonBuilder().serializeNulls().create();
+                    @Nullable
+                    MiIoBasicDevice devdb = gson.fromJson(deviceMapping, MiIoBasicDevice.class);
+                    if (devdb != null) {
+                        for (MiIoBasicChannel channel : devdb.getDevice().getChannels()) {
+                            i18nEntries.put("ch." + key + channel.getChannel(), channel.getFriendlyName());
+                            StateDescriptionDTO state = channel.getStateDescription();
+                            if (state != null) {
+                                List<OptionsValueListDTO> options = state.getOptions();
+                                if (options != null) {
+                                    for (OptionsValueListDTO channelOption : options) {
+                                        String optionValue = channelOption.value;
+                                        String optionLabel = channelOption.label;
+                                        if (optionValue != null && optionLabel != null) {
+                                            i18nEntries.put("option." + key + channel.getChannel() + "-" + optionValue,
+                                                    optionLabel);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                } catch (Exception e) {
+                    LOGGER.info("Error while searching  in database '{}': {}", file.getName(), e.getMessage());
+                }
+            }
+        }
```

Comment:
Consider if you can reduce nesting by inverting some conditions and introducing `continue`, for example.

Reply:
sorry, I don't quite get how this can be done.
I can reduce , but than introduce SAT warnings or need to catch NPE's

Reply:
I was thinking something like changing this:

```
for (File file : filesList) {
    if (file.isFile() && file.getName().toLowerCase().endsWith(".json")) {
        [...]
    }
}
```

to:
```
for (File file : filesList) {
    if (!file.isFile() || !file.getName().toLowerCase().endsWith(".json")) {
        continue;
    }
    [...]
}
```

Reply:
ok, got it.. was looking for improving the ugly nesting with many null checks. Yes, this takes out one level.
I found that I can do the filtering even better, by having the filter in the listfiles, so no need to check it anymore in the loop

Reply:
Was considering other places too, I'll put individual comments to be more explicit about suggestion.